xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1223,7 +1223,8 @@ DemuxPacket* CDVDDemuxFFmpeg::Read()
     else if (stream->type == STREAM_VIDEO)
     {
       if (static_cast<CDemuxStreamVideo*>(stream)->iWidth != m_pFormatContext->streams[pPacket->iStreamId]->codecpar->width ||
-          static_cast<CDemuxStreamVideo*>(stream)->iHeight != m_pFormatContext->streams[pPacket->iStreamId]->codecpar->height)
+          static_cast<CDemuxStreamVideo*>(stream)->iHeight != m_pFormatContext->streams[pPacket->iStreamId]->codecpar->height ||
+		  (stream->disabled && stream->ExtraSize != m_pFormatContext->streams[pPacket->iStreamId]->codecpar->extradata_size))
```

Comment:
@lrusak what you think to make as:
```cpp
      if (!stream->disabled && (static_cast<CDemuxStreamVideo*>(stream)->iWidth != m_pFormatContext->streams[pPacket->iStreamId]->codecpar->width ||
          static_cast<CDemuxStreamVideo*>(stream)->iHeight != m_pFormatContext->streams[pPacket->iStreamId]->codecpar->height) ||
		  stream->ExtraSize != m_pFormatContext->streams[pPacket->iStreamId]->codecpar->extradata_size)
```
Or to have more short and less in mem works:
```cpp
      const auto codecpar = m_pFormatContext->streams[pPacket->iStreamId]->codecpar;
      const auto videostream = static_cast<CDemuxStreamVideo*>(stream);
      if (!stream->disabled && (videostream->iWidth != codecpar->width ||
          videostream->iHeight != codecpar->height) ||
	  videostream->ExtraSize != codecpar->extradata_size)
```

Due to missing compiler not tested now!

Reply:
Looked in the whole function, about the `codecpar`, maybe the best on begin of "if":
```cpp
  // check streams, can we make this a bit more simple?
  if (pPacket->iStreamId >= 0)
  {
    const auto codecpar = m_pFormatContext->streams[pPacket->iStreamId]->codecpar;
    CDemuxStream* stream = GetStream(pPacket->iStreamId);
    ...
```
and then maybe better as separate request?